src/Controller/MessageController.php

```diff
@@ -23,35 +24,45 @@ class MessageController extends AbstractController
     /**
      * TODO: cover this method with tests, and refactor the code (including other files that need to be refactored)
      */
-    #[Route('/messages')]
+    #[Route('/messages', name: 'app_message_list', methods: ['GET'])]
     public function list(Request $request, MessageRepository $messages): Response
     {
-        $messages = $messages->by($request);
-  
-        foreach ($messages as $key=>$message) {
```

Comment:
Too wrong. Arrays should never be modified like that, new variable should have been used. Array keys also should never be used, unless as array struct (or type alias in other languages)